pkg/controller/tidb_control.go

```diff
@@ -137,6 +140,23 @@ func (c *defaultTiDBControl) GetSettings(tc *v1alpha1.TidbCluster, ordinal int32
 	return &info, nil
 }
 
+// SetServerLabels update TiDB's labels config
+func (c *defaultTiDBControl) SetServerLabels(tc *v1alpha1.TidbCluster, ordinal int32, labels map[string]string) error {
+	httpClient, err := c.getHTTPClient(tc)
+	if err != nil {
+		return err
+	}
+
+	buffer := bytes.NewBuffer(nil)
+	if err := json.NewEncoder(buffer).Encode(labels); err != nil {
+		return fmt.Errorf("encode labels to json failed, error: %v", err)
+	}
+
+	url := fmt.Sprintf("%s/labels", c.getBaseURL(tc, ordinal))
```

Comment:
Does it works?

Reply:
The PR in tidb side https://github.com/pingcap/tidb/pull/36845 is merged.